CONTRIBUTING.md

```diff
@@ -0,0 +1,91 @@
+# Contributing to Starling
+
+Hi! We'd love to have your contributions! If you want help or mentorship, reach
+out to us in a GitHub issue, or ping `fitzgen` in
+[#servo on irc.mozilla.org](irc://irc.mozilla.org#servo) and introduce yourself.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Code of Conduct](#code-of-conduct)
+- [Building](#building)
+- [Testing](#testing)
+- [Pull Requests and Code Review](#pull-requests-and-code-review)
+- [Automatic Code Formatting](#automatic-code-formatting)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Code of Conduct
+
+We abide by the [Rust Code of Conduct][coc] and ask that you do as well.
+
+[coc]: https://www.rust-lang.org/en-US/conduct.html
+
+## Building
+
+To build Starling, you need `autoconf` 2.13 and `libclang` at least 3.9. You
+should set the `LIBCLANG_PATH` environment variable to the directory containing
+`libclang.so`.
+
+For example, in Ubuntu 16.04:
+
+```
+$ sudo apt-get install autoconf2.13 clang-4.0
+$ sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-4.0 100
+$ sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-4.0 100
+$ LIBCLANG_PATH=/usr/lib/llvm-4.0/lib cargo build
+```
+
+The `LIBCLANG_PATH` environment variable is only needed by codegen when
+compiling `mozjs`.  Once it is built, you can use the usual cargo commands such
+as `cargo +nightly test`.
+
+## Testing
+
+To run all the tests:
+
+```
+$ cargo test
+```
+
+## Pull Requests and Code Review
+
+Ensure that each commit stands alone, and passes tests. This enables better `git
+bisect`ing when needed. If your commits do not stand on their own, then rebase
+them on top of the latest master and squash them into a single commit.
+
+All pull requests undergo code review before merging. To request review, comment
+`r? @github_username_of_reviewer`. They we will respond with `r+` to approve the
+pull request, or may leave feedback and request changes to the pull request. Any
+changes should be squashed into the original commit.
```

Comment:
Does this work well for bindgen? (I'm assuming you're using it there.) I'd much rather have people push changes addressing feedback as additional patches and then squash them before merging. Then again that makes things quite a bit more complicated.